bionic/persistence.py

```diff
@@ -590,13 +599,27 @@ def _exact_metadata_url_for_query(self, query):
         filename = f"metadata_{query.provenance.exact_hash}.yaml"
         return self._nominal_metadata_url_prefix_for_query(query) + "/" + filename
 
-    def _load_metadata_from_url(self, url):
+    def _load_metadata_if_valid_else_delete(self, url):
```

Comment:
I tried it with a shorter name first. However, then the code in `find_entry` looked like it would loop indefinitely, so I had to add a comment explaining that this function also deletes the URL if it's invalid. At that point I realized that the deletion is a key part of what this function does and the shorter name was (IMO) not an accurate description.